ludwig/features/category_feature.py

```diff
@@ -234,8 +243,16 @@ def logits(self, inputs, **kwargs):  # hidden
         # hidden: shape [batch_size, size of final fully connected layer]
         return {LOGITS: self.decoder_obj(hidden), PROJECTION_INPUT: hidden}
 
+    def create_calibration_module(self, feature) -> torch.nn.Module:
+        calibration_type = feature.get("calibration")
+        if calibration_type == "temperature_scaling":
+            return calibration.TemperatureScaling(num_classes=feature["num_classes"])
+        elif calibration_type == "matrix_scaling":
+            return calibration.MatrixScaling(num_classes=feature["num_classes"])
+        return None
```

Comment:
here and in the binary feature too we can create a registry like we do for most other things and use the functions to get values from registries because they already deal with wrong keys by returning errors. That would help 1) consistency and 2) avoiding silent errors like misspelled calibration type names.